### ISSUE TITLE
Implement series-as-features transformers

### ISSUE THREAD

**Issue by mloning** [alan-turing-institute/sktime#297]
**List of transformations we'd like to include in sktime**
see #6 and #224 

**Describe the solution you'd like**
Here's a template for a series-as-features transformer:

```python
from sktime.transformers.series_as_features.base import \
    BaseSeriesAsFeaturesTransformer
from sktime.utils.validation.series_as_features import check_X

class MyTransformer(BaseSeriesAsFeaturesTransformer):
    """MyTransformer docstring
    """

    def __init__(self, my_parameter=10):
        self.my_parameter = my_parameter
        super(MyTransformer, self).__init__()

    def fit(self, X, y=None):
        """
        Fit transformer.

        Parameters
        ----------
        X : pandas DataFrame of shape [n_samples, n_features]
            Input data
        y : pandas Series, shape (n_samples, ...), optional
            Targets for supervised learning.

        Returns
        -------
        self : an instance of self.
        """
        X = check_X(X, enforce_univariate=True)
        # fit transformer
        ...

        self._is_fitted = True
        return self

    def transform(self, X, y=None):
        """
        Transform X.

        Parameters
        ----------
        X : nested pandas DataFrame of shape [n_instances, n_columns]
            Nested dataframe with time-series in cells.

        Returns
        -------
        Xt : pandas DataFrame
        """
        self.check_is_fitted()
        X = check_X(X)

        # Apply transformation on X
        Xt = ...
        return Xt
```

**Comment by Cheukting** [comment]
I think I get the input but not sure what to be expected for the return Xt? should it be aggregated or flatten out the time series? Also, is this going to be helping out the issue mentioned? Cannot see the connection there... can you explain a bit more?

**Comment by Cheukting** [comment]
Or do you mean by continuing to add the transformer in https://github.com/alan-turing-institute/sktime/issues/6#issuecomment-458490088 ?

**Comment by mloning** [comment]
The output `Xt` would be the same format as the input: a nested `pd.DataFrame`. Yes, so for example, we would like to have a `FourierTransformer()` which takes in a nested data frame with series in cells and returns a nested data frame with Fourier coefficients in cells (applying the Fourier transform iteratively over all series). Hope this helps!

**Comment by ABostrom** [comment]
@mloning hi markus. I'm going to pick up some of the issues. Was thinking of doing some work on the transformers list. Getting my dev env r-synced. looking at the truncation transformer right now.

**Comment by mloning** [comment]
@ABostrom sounds good!

**Comment by mloning** [comment]
Closed this in favour of #483
